### DISCUSSION TITLE
How to define a custom type for graphql

### DISCUSSION THREAD

**Discussion by hqlinh796** [saleor/saleor#11872] [Deployment] [answered]
I want to define a custom Type like "AddressInput" on the Storefront app. How can I do that?

**Comment by peelar** [comment]
Hello @hqlinh796 👋 

Can you go step by step and explain what your problem is? When you reference `AddressInput`, is [this](https://docs.saleor.io/docs/3.x/api-reference/inputs/address-input) what you mean?

**Comment by hqlinh796** [comment]
Yes, I am. I want to add another custom input type. It has some fields which I need. For example Type `CustomAddressInput` which has an additional `isCompany` field.

**Comment by timuric** [comment, marked as answer]
@hqlinh796 Forking Saleor is not recommended for the following reasons:

- Non trivial upgrades due to merge conflicts, incompatible business logic and data migrations.
- Limited or non existing community support.
- Maintenance and security fixes must be applied manually.
- You would have to fork and maintain dashboard which is another large codebase.

We advise you to use metadata fields and webhooks instead.
